src/openrct2/rct2/T6Importer.cpp

```diff
@@ -200,8 +206,30 @@ class TD6Importer final : public ITrackImporter
         }
 
         td->name = _name;
+
+        UpdateRideType(td);
+
         return td;
     }
+
+    void UpdateRideType(std::unique_ptr<TrackDesign>& td)
```

Comment:
Why are you passing in the unique ptr just pass in a track design reference. This function isn't messing about with any memory allocation.

Reply:
I was asked to do so. See the team chat.